gems/aws-sdk-core/lib/aws-sdk-core/plugins/retry_errors.rb

```diff
@@ -196,6 +225,11 @@ def call(context)
           error_inspector = Retries::ErrorInspector.new(response.error, response.context.http_response.status_code)
 
           request_bookkeeping(context, response, error_inspector)
+
+          # Clock skew needs to be updated from the response even when
+          # the request is not retryable
+          config.clock_skew.update_clock_skew(context) if error_inspector.clock_skew?(context)
```

Comment:
You may not like this, but when `ErrorInspector` checks clock skew, we should potentially do the updating there, similar to endpoint discovery. Or even that can be refactored out to something like this.

Reply:
I strongly disagree that `clock_skew?` should do any updating.  I'd prefer to extract the logic for updating the end point cache from the Error Inspector.

Reply:
I agree that the predicate method shouldn't update anything. If we can refactor that out (at least for the retries changes) that would be nice.